performance/benchmarks/benchmark_interpolator.hh

```diff
@@ -0,0 +1,597 @@
+                                         const json & data) const override {
+      auto range = this->template lookup<std::pair<double, double>>(
+          data, "ranges", state);
+      double new_x1 = std::get<0>(range);
+      double new_x2 = std::get<1>(range);
+      int new_log_error_bound =
+          this->template lookup<int>(data, "log_error_bounds", state);
+      auto new_func_name =
+          this->template lookup<SupportedFunc>(data, "func_names", state);
+      return (new_x1 != this->x1 || new_x2 != this->x2 ||
+              new_log_error_bound != this->log_error_bound ||
+              new_func_name != this->func_name);
+    }
+
+    void init_interpolator(const ::benchmark::State & state,
+                           const json & data) override {
+      auto range = this->template lookup<std::pair<double, double>>(
+          data, "ranges", state);
+      this->x1 = std::get<0>(range);
+      this->x2 = std::get<1>(range);
+      this->log_error_bound =
+          this->template lookup<int>(data, "log_error_bounds", state);
+      this->func_name =
+          this->template lookup<SupportedFunc>(data, "func_names", state);
+
+      this->error_bound = std::pow(10, this->log_error_bound);
+      this->init_function();
+      this->intp = std::make_shared<Interpolator_t>(
+          this->func, this->x1, this->x2, this->error_bound);
+    }
+
+    void init_hyp1f1_function() {
+      double n = 10;
+      double l = 9;
+      double a = 0.5 * (n + l + 3);
+      double b = l + 1.5;
+      // Important hyp1f1 a!=b, because for a==b the Hyp1f1 can be simplified
+      // and computations does not take as long as it would on average
+      auto hyp1f1 = math::Hyp1f1(a, b + 1, 200, 1e-15);
+      this->func = [=](double x) mutable { return hyp1f1.calc(x); };
+    }
+
+    void init_function() {
+      switch (this->func_name) {
+      case SupportedFunc::Identity:
+        this->func = [](double x) { return x; };
+        break;
+      case SupportedFunc::Gaussian:
+        this->func = [](double x) {
+          return std::exp(-std::pow((x - 1) / 0.5, 2) / 2);
+        };
+        break;
+      case SupportedFunc::Hyp1f1:
+        this->init_hyp1f1_function();
+        break;
+      default:
+        this->func = [](double x) { return x; };
+        break;
+      }
+    }
+  };
+
+  /**
+   * This class uses the Mat
+   * RadialContribution. Therefore benchmarks for different atomic structures
+   * can be done.
+   */
+  template <class Dataset>
+  class InterpolatorMatrixBFixture : public InterpolatorBFixture<Dataset> {
+   public:
+    using Parent = InterpolatorBFixture<Dataset>;
+    using SupportedVecFunc = typename Dataset::SupportedVecFunc;
+    using Interpolator_t = math::InterpolatorMatrixUniformCubicSpline<
+        math::RefinementMethod_t::Exponential>;
+
+    std::shared_ptr<Interpolator_t> intp{};
+    std::function<math::Matrix_t(double)> func;
+    SupportedVecFunc func_name;
```

Comment:
`std::function<math::Matrix_t(double)> func;` uses default initialization, so `func` is `nullptr` here. For consistency, this could be written as `std::function<math::Matrix_t(double)> func{};`